beacon-chain/powchain/service.go

```diff
@@ -448,6 +456,45 @@ func (s *Service) processSubscribedHeaders(header *gethTypes.Header) {
 	}
 }
 
+// batchRequestHeaders requests the block range specified in the arguments. Instead of requesting
+// each block in one call, it batches all requests into a single rpc call.
+func (s *Service) batchRequestHeaders(startBlock uint64, endBlock uint64) ([]*gethTypes.Header, error) {
```

Comment:
Should we add a test for this?

Reply:
this is actually difficult to test for. We use the raw rpc call over here to batch request all blocks here, so I can't test it out using the simulated backend.

Reply:
Fair 👍🏼